src/ApiHandler.ts

```diff
@@ -361,47 +361,65 @@ export default class ApiHandler {
 		}
 	}
 
+	/**
+	 *
+	 * @param hash BlockHash to make call at.
+	 * @param stash _Stash_ address.
+	 */
 	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-	async fetchStakingLedger(hash: BlockHash, address: string) {
+	async fetchStakingInfo(hash: BlockHash, stash: string) {
 		const api = await this.ensureMeta(hash);
 
-		const [header, staking] = await Promise.all([
+		const [header, controllerOption] = await Promise.all([
 			api.rpc.chain.getHeader(hash),
-			api.query.staking.ledger.at(hash, address),
+			api.query.staking.bonded.at(hash, stash), // Option<AccountId> representing the controller
 		]);
 
 		const at = {
 			hash,
 			height: header.number.toNumber().toString(10),
 		};
 
-		if (staking.isNone) {
-			// Address is not a Controller, no need to look up slashing spans.
-			return {
-				at,
-				staking: {},
-				numSlashingSpans: null,
+		if (controllerOption.isNone) {
+			// if controllerOption.isNone, `stash` address param is not a stash.
+			throw {
+				error: `The address ${stash} is not a stash address.`,
+				statusCode: 400,
 			};
 		}
 
-		const ledger = staking.unwrap(); // should always work if staking.isSome
-		const slashingSpans = await api.query.staking.slashingSpans.at(
-			hash,
-			ledger.stash
-		);
+		const controller = controllerOption.unwrap();
 
-		let numSlashingSpans;
-		if (slashingSpans.isSome) {
-			const span = slashingSpans.unwrap();
-			numSlashingSpans = span.prior.length + 1;
-		} else {
-			numSlashingSpans = 0;
+		const [
+			stakingLedgerOption,
+			rewardDestination,
+			slashingSpansOption,
+		] = await Promise.all([
+			await api.query.staking.ledger.at(hash, controller),
+			await api.query.staking.payee.at(hash, stash),
+			await api.query.staking.slashingSpans.at(hash, stash),
+		]);
+
+		const stakingLedger = stakingLedgerOption.unwrapOr(null);
+
+		if (stakingLedger === null) {
+			// should never throw because by time we get here we know we have a bonded pair
+			throw {
+				error: `Staking ledger could not be found for address "${controller.toString()}"`,
```

Comment:
Or just say stash, since that's what the user passed in.
```suggestion
				error: `Staking ledger could not be found for controller address "${controller.toString()}"`,
```